crates/datagen/src/lib.rs

```diff
@@ -1825,3 +2028,37 @@ impl<'a> RecordGenerator<'a> {
         Ok(self.json_obj.as_ref().unwrap())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{
+        decimal_max, decimal_min, FELDERA_MAX_DECIMAL_PRECISION, FELDERA_MAX_DECIMAL_SCALE,
+    };
+    use rust_decimal::Decimal;
+
+    /// Verify we can generate all feldera min/max decimals without panicking.
+    #[test]
+    fn rust_decimal_min_max_no_panics() {
+        for p in 0..=FELDERA_MAX_DECIMAL_PRECISION {
+            for s in 0..=FELDERA_MAX_DECIMAL_SCALE {
+                if p >= s {
```

Comment:
This looks weird, why not just do `for s in 0..=p` instead of generating the bigger values and skipping them? Or maybe you're concerned that max scale can be less than max precision? But I don't think that's how our library works.